pkg/distro/fedora/images.go

```diff
@@ -639,6 +639,80 @@ func iotSimplifiedInstallerImage(workload workload.Workload,
 	return img, nil
 }
 
+func coreosImage(workload workload.Workload,
+	t *imageType,
+	bp *blueprint.Blueprint,
+	options distro.ImageOptions,
+	_ map[string]rpmmd.PackageSet,
+	_ []container.SourceSpec,
+	rng *rand.Rand) (image.ImageKind, error) {
+
+	container, ref, err := makeOSTreePayloadContainer(options.OSTree, t.OSTreeRef())
+	if err != nil {
+		return nil, fmt.Errorf("%s: %s", t.Name(), err.Error())
+	}
+	img := image.NewOSTreeContainerDiskImage(container, ref)
+
+	distro := t.Arch().Distro()
+
+	customizations := bp.Customizations
+	img.Users = users.UsersFromBP(customizations.GetUsers())
+	img.Groups = users.GroupsFromBP(customizations.GetGroups())
+
+	img.Directories, err = blueprint.DirectoryCustomizationsToFsNodeDirectories(customizations.GetDirectories())
+	if err != nil {
+		return nil, err
+	}
+	img.Files, err = blueprint.FileCustomizationsToFsNodeFiles(customizations.GetFiles())
+	if err != nil {
+		return nil, err
+	}
+
+	img.KernelOptionsAppend = []string{
+		"rw",
+		"console=tty0",
+		"console=ttyS0",
+	}
+
+	img.SysrootReadOnly = true
+
+	img.Platform = t.platform
+	img.Workload = workload
+
+	img.OSName = "fedora-coreos"
+	img.Remote = ostree.Remote{
+		Name: "fedora",
+	}
+
+	if !common.VersionLessThan(distro.Releasever(), "38") {
+		switch img.Platform.GetImageFormat() {
+		case platform.FORMAT_RAW:
+			img.IgnitionPlatform = "metal"
+			if bpIgnition := customizations.GetIgnition(); bpIgnition != nil && bpIgnition.FirstBoot != nil && bpIgnition.FirstBoot.ProvisioningURL != "" {
+				img.KernelOptionsAppend = append(img.KernelOptionsAppend, "ignition.config.url="+bpIgnition.FirstBoot.ProvisioningURL)
+			}
+		case platform.FORMAT_QCOW2:
+			img.IgnitionPlatform = "qemu"
+		}
+	}
+
+	if kopts := customizations.GetKernel(); kopts != nil && kopts.Append != "" {
+		img.KernelOptionsAppend = append(img.KernelOptionsAppend, kopts.Append)
+	}
+
+	// TODO: move generation into LiveImage
+	pt, err := t.getPartitionTable(customizations.GetFilesystems(), options, rng)
+	if err != nil {
+		return nil, err
+	}
+	img.PartitionTable = pt
+
+	img.Filename = t.Filename()
+	img.Compression = t.compression
+
+	return img, nil
+}
```

Comment:
There is a lot going on here. Maybe @achilleas-k we can get together and step through some of it.

Reply:
Sure, we can do that.  For a high level description though: Almost all of this function is essentially creating the `img`
```go
img := image.NewOSTreeContainerDiskImage(container, ref)
```
followed by setting options on the image based on image type settings and user customizations.